aptos-node/src/state_sync.rs

```diff
@@ -64,9 +65,21 @@ pub fn create_event_subscription_service(
         .subscribe_to_reconfigurations()
         .expect("Mempool must subscribe to reconfigurations");
 
-    // Create a reconfiguration subscription for consensus (if this is a validator)
+    // Create a reconfiguration subscription for consensus observer (if enabled)
+    let consensus_observer_reconfig_subscription =
+        if node_config.consensus_observer.observer_enabled {
+            Some(
+                event_subscription_service
+                    .subscribe_to_reconfigurations()
+                    .expect("Consensus observer must subscribe to reconfigurations"),
+            )
+        } else {
+            None
+        };
+
+    // Create a reconfiguration subscription for consensus
     let consensus_reconfig_subscription = if node_config.base.role.is_validator()
-        || node_config.state_sync.state_sync_driver.observer_enabled
+        || node_config.consensus_observer.observer_enabled
```

Comment:
we need to remove this line, it creates a duplicate subscription that's being dropped

Reply:
Aah! Missed it 😄 Thanks @zekun000!